### PR TITLE
PBS Bid Adapter: add dchain (demand chain object) to prebid server adapter

### PR DESCRIPTION
**Pull request by patmmccann**
## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
solves #6380 for prebid server adapter

### CODE REVIEW

**Comment by smenzer** [comment]
@patmmccann has this solution been agreed to? code looks straightforward enough but I didn't see any comments in the Issue so I'm not sure if it requires discussion/approval before being merged.  

cc @gglas in case i'm missing any context

**Comment by patmmccann** [comment]
> @patmmccann has this solution been agreed to? code looks straightforward enough but I didn't see any comments in the Issue so I'm not sure if it requires discussion/approval before being merged.
> 
> cc @gglas in case i'm missing any context

#6380 was adopted as noncontroversial in yesterday's prebid js meeting

**Comment by smenzer** [comment]
> #6380 was adopted as noncontroversial in yesterday's prebid js meeting; this pr was requested during the discussion. also tagging @bretg who agreed to take a look at it. Thanks Bret and Scott!

Ok thanks ... I missed the meeting due to a conflict, so thanks for sharing.

File: modules/prebidServerBidAdapter/index.js
Comment (bretg) [inline modules/prebidServerBidAdapter/index.js:933]: Based on the unit test below, dchain is an object, not a stringified object, so I think there's a problem here.
This isn't a deep copy -- believe the right way to copy a full object would be something like this:

```
Object.assign(bidObject.meta.dchain, bid.ext.dchain);
```

@ChrisHuie - please confirm?

Comment (patmmccann) [reply modules/prebidServerBidAdapter/index.js:933]: happy to make that change, but i'm curious why my test would have passed if this is a problem

Comment (ChrisHuie) [reply modules/prebidServerBidAdapter/index.js]: I think I am fully understanding what you are asking. Object.assign doesn't exactly deep copy but it does create a new clone from the source object that copies over the properties and is then allocated differently in memory. The "=" sign, in this case, instead is referencing the source property which acts just like a pointer to the same memory slot. So if bidObject.meta.dchain changes then bid.ext.dchain would also change and vice-versa I believe. With object.assign the changed source property wouldn't affect bid.ext.dchain.

I think object.assign is better for the side effect reason but shouldn't impact these tests

Comment (patmmccann) [reply modules/prebidServerBidAdapter/index.js]: i changed to utils.deepClone

**Review by bretg — APPROVED** [review]

**Review by ChrisHuie — APPROVED** [review]

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/2737

**Pull request merged**
